libs/proxy/tokio-postgres2/src/connect.rs

```diff
@@ -25,13 +27,14 @@ where
         .make_tls_connect(hostname)
         .map_err(|e| Error::tls(e.into()))?;
 
-    match connect_once(&config.host, config.port, tls, config).await {
+    match connect_once(config.host_addr, &config.host, config.port, tls, config).await {
         Ok((client, connection)) => Ok((client, connection)),
         Err(e) => Err(e),
     }
 }
 
 async fn connect_once<T>(
+    host_addr: Option<IpAddr>,
```

Comment:
nit: The connect functions could now take an Either<IpAddr, Host> since they currently ignore host when ipaddr is passed. This makes it clear that only one is needed.

Reply:
It still needs it for the `SocketConfig` that is returned, this is used in cancellation, which also needs the hostname for TLS.